contracts/test/strategies/sonic/swapx-amo.sonic.fork-test.js

```diff
@@ -0,0 +1,1234 @@
+    const vaultAssetsAfter = await oSonicVault.totalValue();
+    const profit = vaultAssetsAfter
+      .sub(dataBefore.vaultAssets)
+      .add(dataBefore.osSupply.sub(osSupplyAfter));
+
+    log(
+      `Change strat balance: ${formatUnits(
+        stratBalanceAfter.sub(dataBefore.stratBalance)
+      )}`
+    );
+    log(
+      `Change vault assets : ${formatUnits(
+        vaultAssetsAfter.sub(dataBefore.vaultAssets)
+      )}`
+    );
+    log(
+      `Change OS supply    : ${formatUnits(
+        osSupplyAfter.sub(dataBefore.osSupply)
+      )}`
+    );
+    log(`Profit              : ${formatUnits(profit)}`);
+
+    return profit;
+  };
+
+  const assertChangedData = async (dataBefore, delta) => {
+    const { oSonic, oSonicVault, swapXAMOStrategy, swapXPool, swapXGauge, wS } =
+      fixture;
+
+    if (delta.stratBalance != undefined) {
+      const expectedStratBalance = dataBefore.stratBalance.add(
+        delta.stratBalance
+      );
+      log(`Expected strategy balance: ${formatUnits(expectedStratBalance)}`);
+      expect(await swapXAMOStrategy.checkBalance(wS.address)).to.withinRange(
+        expectedStratBalance.sub(3),
+        expectedStratBalance.add(3),
+        "Strategy's check balance"
+      );
+    }
+
+    if (delta.osSupply != undefined) {
+      const expectedSupply = dataBefore.osSupply.add(delta.osSupply);
+      expect(await oSonic.totalSupply(), "OSonic total supply").to.equal(
+        expectedSupply
+      );
+    }
+
+    // Check Vault's wS balance
+    if (delta.vaultWSBalance != undefined) {
+      expect(await wS.balanceOf(oSonicVault.address)).to.equal(
+        dataBefore.vaultWSBalance.add(delta.vaultWSBalance),
+        "Vault's wS balance"
+      );
+    }
+
+    // Check the pool's reserves
+    if (delta.reserves != undefined) {
+      const { _reserve0: wsReserves, _reserve1: osReserves } =
+        await swapXPool.getReserves();
+
+      // If the wS reserves delta is a function, call it to check the wS reserves
+      if (typeof delta.reserves.ws == "function") {
+        // Call test function to check the wS reserves
+        delta.reserves.ws(wsReserves);
+      } else {
+        expect(wsReserves, "wS reserves").to.equal(
+          dataBefore.reserves.ws.add(delta.reserves.ws)
+        );
+      }
+      // Check OS reserves delta
+      expect(osReserves, "OS reserves").to.equal(
+        dataBefore.reserves.os.add(delta.reserves.os)
+      );
+    }
+
+    if (delta.stratGaugeBalance) {
+      // Check the strategy's gauge balance
+      const expectedStratGaugeBalance = dataBefore.stratGaugeBalance.add(
+        delta.stratGaugeBalance
+      );
+      expect(
+        await swapXGauge.balanceOf(swapXAMOStrategy.address)
+      ).to.withinRange(
+        expectedStratGaugeBalance.sub(1),
+        expectedStratGaugeBalance.add(1),
+        "Strategy's gauge balance"
+      );
+    }
+  };
+
+  async function assertDeposit(wsDepositAmount) {
```

Comment:
many of these asserts could also test that after the operation the LP token balance of the strategy contract equals 0

Reply:
and they could also assert that the strategy has a 0 OS balance after any operation

Reply:
That's a good suggestion. I've added this to the deposit, withdraw and swap assert functions.